language-design.md

```diff
@@ -17,7 +17,78 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
-
+> I have said in the past, and will say now, that I think it would be a good thing
+> for the Java programming language to ... let the user define overloaded operators.
+> Just as a user can code methods that can be used in just the same way as
+> methods that are built in, the user ought to have a way to define operators for user
+> defined classes that can be used in just the same way as operators that are built in. [Steele, 1998]
+
+I never really considered the importance of operator overloading before, but Steele makes a very convincing argument for them:
+opertor overloading allows an extension of a language (e.g., a class or package) to be as concise, familiar, and 'primitive'
+as the underlying language. For example, in languages such as Java where operator overloading is not supported,
+users must write an `add` function to represent the addition of two objects; the resulting code looks something like
+`add(obj1, obj2)`, which gets the job done but isn't as simple and pleasing as the operator overloaded alternative `obj1 + obj2`.
```

Comment:
This is incredibly annoying in Java, especially when you consider symmetry.  Suppose you want to make your own string class that does some extra stuff.  You can't subclass `String` in Java because it's marked final, which ruins symmetry

```
MyStringClass myInstance = new MyStringClass("hello");
String javaInstance = new String("hello");
myInstance.equals(javaInstance); // True
javaInstance.equals(myInstance); // False
```

which is ridiculous - it should be true in both directions.